fedimint-core/src/config.rs

```diff
@@ -231,29 +228,42 @@ impl ClientConfig {
 ///
 /// The same `ModuleKind` may have multiple instances with different settings
 #[derive(Debug, Clone, Default)]
-pub struct ConfigGenParams(BTreeMap<String, serde_json::Value>);
+pub struct ConfigGenParams {
+    modules: BTreeMap<ModuleInstanceId, JsonWithKind>,
+    pub registry: ModuleGenRegistry,
```

Comment:
This being here doesn't look correct. Convenient maybe somewhere, but how is "module gen registry" a part of "config gen params"?

Reply:
Doesn't have to be but it ensures every configured module is also registered, so the developer only needs to do it in one place and doesn't accidentally miss registering a module.

Reply:
This can surely be achieved in some other ways. Again. This `struct` used to mean "extra arguments for a single module instance", and now without changing the name (or even how it used) it is working completely differently.